### PR TITLE
Add e2e region based tests

### PR DESCRIPTION
**Pull request by nickmly**
# Summary

Adds e2e detox tests for the region based screens (AllSetView, NoCode, NoExposureView).
In order to do this, DemoMode menu items were added to force the NoExposureViews.

# Test instructions

1. Run `yarn pre:test:ios` or `yarn pre:test:android` 
2. Run `yarn test:ios` or `yarn test:android`
3. Observe all tests pass

# Help requested

> Things that you (the submitter) want reviewers to pay very close attention to when they review this.

# Unresolved questions / Out of scope

> Are there any related issues or tangent features you consider out of scope for this issue that could be addressed in the future?

# Reviewer checklist

This is a suggested checklist of questions reviewers might ask during their review:

- [ ] Does this meet a user need?
- [ ] Is it accessible?
- [ ] Is it translated between both offical languages?
- [ ] Is the code maintainable?
- [ ] Have you tested it?
- [x] Are there automated tests?
- [ ] Does this cause automated test coverage to drop?
- [ ] Does this break existing functionality?
- [ ] Should this be split into smaller PRs to decrease change risk?
- [ ] Does this change the privacy policy?
- [ ] Does this introduce any security concerns?
- [ ] Does this significantly alter performance?
- [ ] What is the risk level of using added dependencies?
- [ ] Should any documentation be updated as a result of this? (i.e. README setup, etc.)

### CODE REVIEW

File: src/testMode/DemoMode.tsx [hunk 1]
```diff
@@ -25,6 +25,10 @@ const ScreenRadioSelector = () => {
   const {forceScreen, setForceScreen} = useStorage();
   const screenData = [
     {displayName: 'None', value: 'None'},
+    {displayName: 'Not Exposed', value: 'NoExposureView'},
+    {displayName: 'Not Exposed (Region covered)', value: 'NoExposureCoveredRegionView'},
+    {displayName: 'Not Exposed (Region not covered)', value: 'NoExposureUncoveredRegionView'},
+    {displayName: 'Not Exposed (Region not set)', value: 'NoExposureNoRegionView'},
```
Comment (smcmurtry) [inline src/testMode/DemoMode.tsx:31]: It doesn't look like you're actually using these in the detox tests, is that correct? I would prefer if we don't add these so we don't have to expand the functionality of the test menu more than is necessary.

File: e2e/province.e2e.js [hunk 2]
```diff
@@ -64,3 +78,55 @@ describe('Test province flow', () => {
...
+    await changeRegion('AB');
+    await element(by.id('tapPromptCollapsed')).tap();
+    await expect(element(by.id('getCodeButton'))).toBeVisible();
+    await element(by.id('getCodeButton')).tap();
+    await device.takeScreenshot('noCodeAB');
+    await expect(element(by.id('noCodeHeader'))).toBeVisible();
+    await expect(element(by.id('noCodeCTA'))).toBeNotVisible();
+    await element(by.id('toolbarCloseButton')).tap();
+    await element(by.id('bottom-sheet-close')).tap();
+  });
```
Comment (smcmurtry) [inline e2e/province.e2e.js:102]: I think this test will pass even if AB and ON display the same content on the no code screen. Could you check for either different text, or that the button only exists on the ON screen?

Comment (nickmly) [reply e2e/province.e2e.js]: @smcmurtry I fixed up the Android tests and I am using `await expect(element(by.id('noCodeCTA'))).not.toBeVisible();` to determine if the CTA button is visible on the No Code screen, is this sufficient for testing?

Comment (smcmurtry) [reply e2e/province.e2e.js]: Yes that is great!

**Pull request merged**